sonarts-core/scripts/NewRule/src/main.ts

```diff
@@ -0,0 +1,267 @@
+import fs = require("fs");
+
+function showHelp() {
+  console.log("");
+  console.log("Before using the script, first run the rule API from SonarTS folder. For example:");
+  console.log(
+    "java -jar /tools/rule-api-1.17.0.1017.jar generate -language ts -rule S4043 -preserve-filenames -no-language-in-filenames ",
+  );
+  console.log("");
+  console.log("Usage:");
+  console.log("node main.ts RSPEC-KEY className isTypescriptOnly");
+  console.log("");
+  console.log("Example:");
+  console.log("node main.ts S4043 myFirstAwesomeRule true");
+  console.log("");
+}
+
+function escapeRegExp(str: string) {
+  return str.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, "\\$1");
+}
+
+function replace(text: string, dictionary): string {
+  for (let tok in dictionary) {
+    text = text.replace(new RegExp(escapeRegExp(tok), "g"), dictionary[tok]);
+  }
+  return text;
+}
+
+function copyWithReplace(src: string, dest: string, dict) {
+  let content = fs.readFileSync(src).toString();
+  let newContent = replace(content, dict);
+  fs.writeFileSync(dest, newContent);
+}
+
+function createFiles(
+  rootFolder: string,
+  ruleClassName: string,
+  ruleNameDash: string,
+  ruleTitle: string,
+  rspecKey: string,
+  isTypescriptOnly: boolean,
+) {
+  const templatesFolder = `${__dirname}/../templates`;
+  const ruleTemplatePath = `${templatesFolder}/rule_template.ts`;
+
+  let dict = {};

```

Comment:
just :
```
let dict = {ruleNameDash, ruleClassName, ruleTitle, etc. };
```
Also, consider renaming `dict` to something more meaningful, like `ruleMetadata` or something